### PR TITLE
add_gm_smx_openssl3_incompatible_mode

### PR DESCRIPTION
**Pull request by xiaonan-INTC**
add support for GM SMx sample project for OpenSSL 3.0 incompatible mode

### CODE REVIEW

File: Linux/sgx/test_app/enclave/tests/evp_smx.c [hunk 1]
```diff
@@ -33,14 +33,12 @@
 #include <stdio.h> /* vsnprintf */
 #include <string.h>
 
-#include "openssl/bio.h"
+#include "openssl/bio.h" // #include "internal/bio.h"
```
Comment (llly) [inline Linux/sgx/test_app/enclave/tests/evp_smx.c:36]: unnecessary comment

File: Linux/sgx/test_app/enclave/tests/evp_smx.c [hunk 2]
```diff
@@ -33,14 +33,12 @@
...
 #include "openssl/evp.h"
 #include "openssl/ec.h"
 #include "openssl/pem.h"
-#if OPENSSL_VERSION_NUMBER >= 0x30000000
 #include "openssl/decoder.h"
-#include "openssl/core_names.h"
-#endif /* openssl library */
+#include "openssl/core_names.h" /* openssl library */
```
Comment (llly) [inline Linux/sgx/test_app/enclave/tests/evp_smx.c:41]: unnecessary comment

**Comment by jinghe-INTC** [comment]
Please change your target branch to https://github.com/intel/intel-sgx-ssl/tree/support_openssl_3.0-no_deprecated-dev . The master branch is still for 1.1.1.

**Comment by xiaonan-INTC** [comment]
need to change remote pushed branch, will submit new, close it

**Pull request closed without merging**